### DISCUSSION TITLE
Novice user question(s) on GUI

### DISCUSSION THREAD

**Discussion by smhill001** [ortk95/planetmapper#399] [Q&A] [answered]
Hi,

I'm starting to learn and use Planetmapper on a Windows 11 laptop in the Spyder IDE. I typically run all my code from within this IDE. The GUI doc says, "To start, type planetmapper into a command line and press Enter. This will open a window where you can choose a file to open:" I do this on the IDE command line and it simply reports a description of planetmapper:

In[4]: planetmapper
Out[4]: <module 'planetmapper' from 'C:\\Users\\smhil\\anaconda3\\Lib\\site-packages\\planetmapper\\__init__.py'>

Is this intended to be run from the OS command line instead? 

Thanks!

I'll be using amateur data that won't come with WCS data and may not initially be in FITS format (e.g., PNG, JPG).

**Comment by ortk95** [comment]
Hi @smhill001,

Great question! You're right that the `planetmapper` command only works when run from an OS command line, so won't work when run from an interactive Python terminal. The documentation doesn't make this clear at the moment, so I'll update it to make it less confusing - thanks for pointing it out!

Currently, the easiest way to launch the GUI from a Python terminal (or any other Python code) is by running:
```python
planetmapper.gui.GUI().run()
```
This is basically what happens under the hood when you run the `planetmapper` command in an OS command line, so should give you what you want (see below for a couple of slightly more complex alternatives).

This is definitely very cumbersome, so I'll also add an easier way to launch the GUI from a Python terminal in the next release of PlanetMapper. This will probably be something more logical, like `planetmapper.run_gui()`, though the full `planetmapper.gui.GUI().run()` version will obviously still work (see issue #400 for more details).

I hope this helps - let me know if any of this doesn't work/doesn't make sense, or if you have any other questions!

~ Oliver


---

Here are a couple of alternatives for launching the GUI from a python terminal/script - they're a little bit more complex, but might be useful, depending on what you're doing...

If you already know the path to your data file, you could do something like this to launch the GUI with the appropriate file already open ([see docs](https://planetmapper.readthedocs.io/en/latest/documentation.html#planetmapper.Observation.run_gui)):
```python
observation = planetmapper.Observation('/path/to/data.png', target='jupiter', utc='2024-09-18T00:00')
observation.run_gui()
```

If you already have a good idea of some of the disc parameters, you can set these before running the GUI. See [this section of the documentation](https://planetmapper.readthedocs.io/en/latest/documentation.html#planetmapper.BodyXY.set_disc_params) for a list of the various useful methods to set disc parameters:
```python
observation = planetmapper.Observation('/path/to/data.png', target='jupiter', utc='2024-09-18T00:00')
observation.set_rotation(123.45)
observation.set_plate_scale_arcsec(67.89)
observation.run_gui()
```

**Reply by smhill001** [reply]
Thanks! Very helpful and worked perfectly

**Reply by smhill001** [reply]
Follow on question: Now that I have *map.fits files, I want to perform operations on them, 'stack' multiple maps, and then reproject them as disks, e.g., orthographic projections. I played around with BodyXY.plot_map, but couldn't quite sort out what needed to be done without an example. Is it possible to give a quick example of plotting an orthographic projection from a *.map.fits file (rectangular projection)? Thanks again!

**Reply by ortk95** [reply]
That's (currently) something that PlanetMapper isn't able to natively do - at the moment, PlanetMapper can only transform observed images to any arbitrary map projection, but doesn't have the functionality to easily transform from one map projection to another.

If you want to throw some code together to do this yourself, then I think [Cartopy](https://scitools.org.uk/cartopy/docs/latest/) or [pyproj](https://pyproj4.github.io/pyproj/stable/) would probably be the best libraries to use:
- Cartopy is effectively a Matplotlib extension that allows you to relatively easily plot data to various different map projections. For example, you can do something along these lines to plot rectangularly projected data in an orthographic projection:
```python
import cartopy.crs as ccrs
import matplotlib.pyplot as plt

plt.figure()
ax = plt.axes(projection=ccrs.Orthographic())
ax.imshow(data, projection=ccrs.PlateCarree())
plt.show()
```
- If you want to play with the reprojected data yourself, then pyproj is more powerful, but also a bit harder to use (it's basically what both PlanetMapper and cartopy use internally for map projection stuff). I actually have some old example code at https://github.com/ortk95/reproject-data for changing from rectangular to orthographic projections that may be useful.

Reprojecting between map projections would be a really useful feature to add to PlanetMapper, so I've added it to the to-do list! I'm not sure when exactly I'll get it implemented, but you can track the issue at #403 if you're interested.

**Comment by ortk95** [comment, marked as answer]
For completeness, with the latest release of PlanetMapper (v1.12.1), it's now possible to launch the GUI from a Python terminal (or any other Python code) by running:
```python
planetmapper.run_gui()
```
This Python command is directly equivalent to running the `planetmapper` command in an Operating System command line.

See the [GUI tutorial](https://planetmapper.readthedocs.io/en/latest/user_interface.html) and [documentation](https://planetmapper.readthedocs.io/en/latest/documentation.html#planetmapper.run_gui) for more details.
